js/languages.js

```diff
@@ -156,9 +157,9 @@
         // check if we have any matches and they are exist in the array we have
       } else if ((matches && matches[0]) && supportedLanguages.indexOf(matchesLang) !== -1) {
         href = href.replace(matches[0], selectedLang);
-        window.location = href;
+        window.location = href + query;
```

Comment:
What about the URL's `hash` if present?  We should be careful to put this back together in such a way that it will work for all types of URLs.  You can easily have both a `#foo` and `?bar=1` on your URL.